lib/board.rb

```diff
@@ -0,0 +1,46 @@
+require 'pry'
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")  # @cells and self.cells are the same statements (if there is no cells method for your class).
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(user_input)
+    cells[user_input.to_i - 1]
+  end
+
+  def full?
+    cells.all?{|c| c == "X" || c == "O"}
+  end
+
+  def turn_count  #returns the amount of used/played cells
+    cells.count{ |c| c == "X" || c == "O"}  #counts when X or O is on board
+  end
+
+  def taken?(user_input)
+    position(user_input) == " " ? false : true   #position(user_input) == "X" || position(user_input)== "O"
```

Comment:
Avoid using the ternary operator to return `true` and `false`. Instead, rely on the implicit boolean return value of the expression. 

`cells.none? {|cell| cell == " " || cell == nil}`